perf/src/deduper.rs

```diff
@@ -69,28 +70,28 @@ impl<const K: usize> Deduper<K> {
         }
         out
     }
+}
 
-    pub fn dedup_packets_and_count_discards(
-        &self,
-        batches: &mut [PacketBatch],
-        mut process_received_packet: impl FnMut(&mut Packet, bool, bool),
-    ) -> u64 {
-        batches
-            .iter_mut()
-            .flat_map(PacketBatch::iter_mut)
-            .map(|packet| {
-                if packet.meta().discard() {
-                    process_received_packet(packet, true, false);
-                } else if self.dedup_packet(packet) {
-                    packet.meta_mut().set_discard(true);
-                    process_received_packet(packet, false, true);
-                } else {
-                    process_received_packet(packet, false, false);
-                }
-                u64::from(packet.meta().discard())
-            })
-            .sum()
-    }
+pub fn dedup_packets_and_count_discards<const K: usize>(
```

Comment:
this can move to where ever `Packet` and `PacketBatch` are defined nowadays. (future PR)